{
  "WorkItem": {
    "AffectedComponent": {
      "Name": "",
      "DisplayName": ""
    },
    "ClosedComment": "This appears to be fixed in the latest bits in the develop branch. I always see an error on line 21.",
    "ClosedDate": "2013-07-11T17:10:42.907-07:00",
    "CommentCount": 0,
    "Custom": null,
    "Description": "The problem is explained inline.\r\n\r\n``` TypeScript\r\ninterface IPromise<T> {\r\n\tnext<R>(success: (value: T) => R): IPromise<R>\r\n}\r\n\r\nclass Promise<T> implements IPromise<T> {\r\n\tvalue: T;\r\n\t\r\n\tconstructor(value: T) {\r\n\t\tthis.value = value;\r\n\t}\r\n\t\r\n\tnext<R>(success: (value: T) => R): IPromise<R> {\r\n\t\treturn new Promise(success(this.value));\r\n\t}\r\n}\r\n\r\nvar promise = new Promise(42);\r\nfunction success(value: number): number {\r\n\t// The line below should have a type error. `notexisting` is not defined\r\n\t// on a number. Line 21 is just below:\r\n\treturn value.notexiting();\r\n}\r\n\r\n// Comment out this line and the compiler will complain about the type error\r\n// on line 21.\r\nvar next = promise.next(success);\r\n```\r\n\r\n[This code in the playground](http://www.typescriptlang.org/Playground/#src=interface%20IPromise%3CT%3E%20%7B%0D%0A%09next%3CR%3E(success%3A%20(value%3A%20T)%20%3D%3E%20R)%3A%20IPromise%3CR%3E%0D%0A%7D%0D%0A%0D%0Aclass%20Promise%3CT%3E%20implements%20IPromise%3CT%3E%20%7B%0D%0A%09value%3A%20T%3B%0D%0A%09%0D%0A%09constructor(value%3A%20T)%20%7B%0D%0A%09%09this.value%20%3D%20value%3B%0D%0A%09%7D%0D%0A%09%0D%0A%09next%3CR%3E(success%3A%20(value%3A%20T)%20%3D%3E%20R)%3A%20IPromise%3CR%3E%20%7B%0D%0A%09%09return%20new%20Promise(success(this.value))%3B%0D%0A%09%7D%0D%0A%7D%0D%0A%0D%0Avar%20promise%20%3D%20new%20Promise(42)%3B%0D%0Afunction%20success(value%3A%20number)%3A%20number%20%7B%0D%0A%09%2F%2F%20The%20line%20below%20should%20have%20a%20type%20error.%20%60notexisting%60%20is%20not%20defined%0D%0A%09%2F%2F%20on%20a%20number.%0D%0A%09return%20value.notexiting()%3B%0D%0A%7D%0D%0A%0D%0A%2F%2F%20Comment%20out%20this%20line%20and%20the%20compiler%20will%20complain%20about%20the%20type%20error%0D%0A%2F%2F%20on%20line%2021.%0D%0Avar%20next%20%3D%20promise.next(success)%3B) ",
    "LastUpdatedDate": "2013-07-11T17:10:42.907-07:00",
    "PlannedForRelease": "",
    "ReleaseVisibleToPublic": false,
    "Priority": {
      "Name": "Unassigned",
      "Severity": 0,
      "Id": 0
    },
    "ProjectName": "typescript",
    "ReportedDate": "2013-07-10T01:21:10.707-07:00",
    "Status": {
      "Name": "Closed",
      "Id": 4
    },
    "ReasonClosed": {
      "Name": "Fixed"
    },
    "Summary": "Type error with generics not detected",
    "Type": {
      "Name": "Unassigned",
      "Id": 5
    },
    "VoteCount": 2,
    "Id": 1328
  },
  "FileAttachments": [],
  "Comments": []
}